llm/llama.go

```diff
@@ -306,7 +311,9 @@ func newLlama(model string, adapters []string, runners []ModelRunner, numLayers
 		)
 		cmd.Env = append(os.Environ(), fmt.Sprintf("LD_LIBRARY_PATH=%s", filepath.Dir(runner.Path)))
 		cmd.Stdout = os.Stderr
-		cmd.Stderr = os.Stderr
+		var stderr bytes.Buffer
```

Comment:
`bytes.Buffer` isn't ideal since it can grow infinitely if the subprocess continuously writes to Stderr. Instead, you can create a Writer that checks for error and sends to the error channel.

Something like this:

```go
type StatusWriter struct{}

func (w StatusWriter) Write(b []byte) (int, error) {
  if _, after, ok := bytes.Cut(b, "error:"); ok {
    return 0, fmt.Errorf("runner error: %s", after)
  }

  return nil
}
```

Reply:
Great suggestion thanks, this is much nicer.